rl4co/envs/routing/cvrp/local_search.py

```diff
@@ -0,0 +1,162 @@
+        actions: torch.Tensor, Tour indices with shape [batch_size, max_seq_len]
+        max_trials: int, maximum number of trials for local search
+        neighbourhood_params: dict, parameters for neighbourhood search
+        load_penalty: int, penalty for exceeding the vehicle capacity
+        allow_infeasible_solution: bool, whether to allow infeasible solutions
+        seed: int, random seed for local search
+        num_workers: int, number of workers for parallel processing
+    Returns:
+        torch.Tensor, Improved tour indices with shape [batch_size, max_seq_len]
+    """
+
+    # Convert tensors to numpy arrays
+    # Note: to avoid the overhead of device transfer, we recommend to pass the tensors in cpu
+    actions_np = actions.detach().cpu().numpy()
+    positions_np = td["locs"].detach().cpu().numpy()  # [batch_size, num_loc + 1, 2]
+    demands_np = td["demand"].detach().cpu().numpy()  # [batch_size, num_loc]
+    demands_np = np.pad(demands_np, ((0, 0), (1, 0)), mode="constant")  # Add depot demand
+    distances = td.get("distances", None)  # [batch_size, num_loc + 1, num_loc + 1]
+    if distances is None:
+        distances_np = get_distance_matrix(td["locs"]).numpy()
+    else:
+        distances_np = distances.detach().cpu().numpy()
+
+    max_trials = 1 if allow_infeasible_solution else max_trials
+
+    partial_func = partial(
+        local_search_single,
+        neighbourhood_params=neighbourhood_params,
+        load_penalty=load_penalty,
+        allow_infeasible_solution=allow_infeasible_solution,
+        max_trials=max_trials,
+        seed=seed,
+    )
+
+    if num_workers > 1:
+        with Pool(processes=num_workers) as pool:
+            new_actions = pool.starmap(partial_func, zip(actions_np, positions_np, demands_np, distances_np))
+    else:
+        new_actions = [partial_func(*args) for args in zip(actions_np, positions_np, demands_np, distances_np)]
+
+    # padding with zero
+    lengths = [len(act) for act in new_actions] + [actions_np.shape[1]]
+    max_length = max(lengths)
+    new_actions = np.array(
+        [np.pad(act, (0, max_length - length), mode="constant") for act, length in zip(new_actions, lengths)]
+    )
+    return torch.from_numpy(new_actions.astype(np.int64)).to(td.device)
+
+def local_search_single(
+    path: np.ndarray,
+    positions: np.ndarray,
+    demands: np.ndarray,
+    distances: np.ndarray,
+    neighbourhood_params: dict | None = None,
+    allow_infeasible_solution: bool = False,
+    load_penalty: float = 0.2,
+    max_trials: int = 10,
+    seed: int = 0,
+) -> np.ndarray:
+    data = make_data(positions, demands, distances)
+    solution = make_solution(data, path)
+    ls_operator = make_search_operator(data, seed, neighbourhood_params)
+
+    improved_solution, is_feasible = perform_local_search(
+        ls_operator,
+        solution,
+        int(load_penalty * 10**4),  # * 10**4 as we scale the data by 10**4 in `make_data`
```

Comment:
```suggestion
        int(load_penalty * 10**4),  # * 10**4 as we scale the data by 10**4 in `make_data`
```

I suggest making this value a CONSTANT variable and use it `make_data` to. Perhaps also add a comment to the CONSTANT explaining why it's needed to scale the data so that users can understand how to modify if needed.

Reply:
Good point, thanks!